brave/brave-ios
# Fix #2830: Add Night Mode

## Summary of Changes

This PR is adding Night Mode injection to our browser. The script is from Firefox [NightModeHelper](https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentStart/NightModeHelper.js)
and Night Mode is integrated accordingly with Dark/Light Mode.


This pull request fixes #2830

## Submitter Checklist:


## Test Plan:

- Go Settings -> Appearance -> Set APPEARANCE light Mode
- Browse wikipedia.org
- Check Main Page on website is not in night mode and application itself is light mode on
- Go back to Appearance Settings and Set Night mode toggle on 
- Check Application itself goes to dark mode and check open website is in night mode
- Go back to Appearance Settings and Set Night mode toggle off 
- Check Application itself goes to light mode and check open website is not in night mode

So the idea here is When NightMode is enabled it will also effect Application Appearance and it will change application to dark mode and when the Night mode toggle is off, it will switch back to the previous preference. 

## Screenshots:

![test](https://user-images.githubusercontent.com/6643505/157537612-f0cde26f-36a5-4e97-85c1-e68fa42dcf0a.png)

https://user-images.githubusercontent.com/6643505/157545057-ca0addb9-e549-4a61-967b-9976a2352e79.MP4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
